runtime/syntax/ocaml.yaml

```diff
@@ -17,16 +17,16 @@ rules:
     - statement: "\\b(if|then|else)\\b"
       #blocs
     - type: "\\b(begin|end|object|struct|sig|for|while|do|done|to|downto)\\b"
+    - type: "'[0-9A-Za-z_]+"
       #constantes
     - constant.bool: "\\b(true|false)\\b"
       #modules/classes
     - special: "\\b(include|inherit|initializer)\\b"
       #expr modifiers
     - special: "\\b(new|ref|mutable|lazy|assert|raise)\\b"
     - constant.string:
-        start: "'"
-        end: "'"
-        skip: "\\\\."
+        start: "'(.|\\\\(([0-7]{3}|x[A-Fa-f0-9]{2}|u[A-Fa-f0-9]{4}|U[A-Fa-f0-9]{8})|[abfnrtv'\\\"\\\\]))'"
+        end: ""
```

Comment:
Empty `end` is not really well-defined, and in the latest builds micro actually gives a warning about this whenever you load the syntax (since #3256). It might be best to convert the char rule into simple regexes without start/end/rules (although this might cause problems if `\` comes up somewhere else in OCaml syntax – I don't know much about OCaml):

```
    - constant.string: "'(\\\\[0-7]{3}|\\\\x[A-Fa-f0-9]{2}|\\\\u[A-Fa-f0-9]{4}|\\\\U[A-Fa-f0-9]{8}|\\\\[abfnrtv'\\\"\\\\]|.)'"
    - constant.specialChar: "\\\\[abfnrtv'\\\"\\\\]"
    - constant.specialChar: "\\\\([0-7]{3}|x[A-Fa-f0-9]{2}|u[A-Fa-f0-9]{4}|U[A-Fa-f0-9]{8})"
```

Reply:
Oh, thanks! Sorry about that, im using a version from October 2023 and did not realise that. I dont think `\` is used anywhere outside string/character literals, so it should be fine.